qcengine/programs/tests/test_nwchem.py

```diff
@@ -301,3 +301,37 @@ def test_error_correction():
     assert result.success
     assert "geom_binvr" in result.extras["observed_errors"]
     assert result.extras["observed_errors"]["geom_binvr"]["keyword_updates"] == {"geometry__noautoz": True}
+
+
+@pytest.mark.parametrize(
+    "method, keyword, init_iters, use_tce",
+    [
+        ["b3lyp", "dft__maxiter", 4, False],
+        ["b3lyp", "dft__iterations", 4, False],
+        ["hf", "scf__maxiter", 2, False],
+        ["mp2", "scf__maxiter", 2, False],
+        ["mp2", "scf__maxiter", 2, True],
+        ["ccsd", "tce__maxiter", 8, True],
+        ["ccsd", "ccsd__maxiter", 4, False],
+    ],
+)
+@using("nwchem")
+def test_conv_threshold(h20v2, method, keyword, init_iters, use_tce):
+    result = qcng.compute(
+        {
+            "molecule": h20v2,
+            "model": {"method": method, "basis": "sto-3g"},
+            "driver": "energy",
+            "keywords": {
+                keyword: init_iters,
+                "qc_module": use_tce,
```

Comment:
I've been using qc_module = tce/cc/mp2grad/directmp2 rather than a bool here. Can that work with your plans?

Reply:
It would, and I think that gives us better flexibility and clarity that "qc_module" == True means that we're using TCE

Reply:
sounds good. I'll merge as-is unless you've got bool->str edits underway for this PR.

Reply:
I'd merge as-is. Do we have an issue open for the bool->str?

Reply:
no open issue. in #304 I've got the harvester collecting the module and storing it in `provenance.module`. that's a catch-all PR that'll be refactored and split before consideration.